hnn_core/cell.py

```diff
@@ -232,60 +277,26 @@ def _create_sections(self, p_secs, topology):
             child_loc = connection[3]
             child_sec.connect(parent_sec, parent_loc, child_loc)
 
-    def build(self, p_secs, p_syn, topology, sect_loc):
-        """Build cell in Neuron.
+    def build(self, align_dipole=None):
+        """Build cell in Neuron and insert dipole if applicable.
 
         Parameters
         ----------
-        p_secs : nested dict
-            Dictionary with keys as section name.
-            p_secs[sec_name] is a dictionary with keys
-            L, diam, Ra, cm, syns and mech.
-            syns is a list specifying the synapses at that section.
-            The properties of syn are specified in p_syn.
-            mech is a dict with keys as the mechanism names. The
-            values are dictionaries with properties of the mechanism.
-        p_syn : dict of dict
-            Keys are name of synaptic mechanism. Each synaptic mechanism
-            has keys for parameters of the mechanism, e.g., 'e', 'tau1',
-            'tau2'.
-        topology : list of list
-            The topology of cell sections. Each element is a list of
-            4 items in the format
-            [parent_sec, parent_loc, child_sec, child_loc] where
-            parent_sec and parent_loc are float between 0 and 1
-            specifying the location in the section to connect and
-            parent_sec and child_sec are names of the connecting
-            sections.
-        sect_loc : dict of list
-            Can have keys 'proximal' or 'distal' each containing
-            names of section locations that are proximal or distal.
-
-        Examples
-        --------
-        >>> p_secs = {
-            'soma':
-            {
-                'L': 39,
-                'diam': 20,
-                'cm': 0.85,
-                'Ra': 200.,
-                'sec_pts': [[0, 0, 0], [0, 39., 0]],
-                'syns': ['ampa', 'gabaa', 'nmda'],
-                'mechs' : {
-                    'ca': {
-                        'gbar_ca': 60
-                    }
-                }
-            }
-        }
+        align_dipole : str | None
```

Comment:
use the same name as in `insert_dipole` ? Also not sure if `insert_dipole` should still be public method according to this change?